### ISSUE TITLE
Maven plugin's `includeGeneratedAnnotation` does not support `jakarta.annotation.Generated`

### ISSUE THREAD

**Issue by anno1985** [joelittlejohn/jsonschema2pojo#1461]
When using both `includeJsr303Annotations` and `includeGeneratedAnnotation`, it seems one still gets `javax.annotation.(processing.)Generated` instead of [`jakarta.annotaiton.Generated`](https://jakarta.ee/specifications/annotations/2.0/annotations-spec-2.0.html#jakarta-annotation-generated).

**Comment by joelittlejohn** [comment]
So I guess javax has become jakarta, and now we have another Generated annotation to throw into the mix :)

**Comment by unkish** [comment]
Current annotations are part of the JRE, thus shouldn't require additional dependencies.
Would throwing in `jakarta.annotation` in the mix mean yet another flag, as both:
`includeJsr303Annotations` and `useJakartaValidation` are about _validation api_:
https://github.com/joelittlejohn/jsonschema2pojo/blob/8e772064776c51376d63c35674e6dd5a0d5e123c/pom.xml#L373-L382

Whereas [jakarta.annotaiton.Generated](https://jakarta.ee/specifications/annotations/2.0/annotations-spec-2.0.html#jakarta-annotation-generated) would be about _annotations_:
```
<dependency>
    <groupId>jakarta.annotation</groupId>
    <artifactId>jakarta.annotation-api</artifactId>
    <version>2.1.1</version>
</dependency>
```

**Comment by joelittlejohn** [comment]
:weary:

**Comment by joelittlejohn** [comment]
It's funny how an annotation is such a trivial thing, just a tag with no behaviour at all, but it's bounced around so many different locations over the years. Why can't they just include this trivial annotation in Java's core API!

**Comment by eirnym** [comment]
I think it's mostly political reasons between Java Core and J2EE.

**Comment by darkmastermindz** [comment]
Hi any updates on this? @joelittlejohn This is blocking JDK17+ compatibility which is required for CVE fixes.

**Comment by GregDThomas** [comment]
FWIW, https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator-gradle-plugin/README.adoc has the same problem (i.e. javax vs jakarta) - and overcomes it with a single flag "useJakartaEe= true/false" to determine if it uses javax.* or jakarta.* style annotations.

**Comment by GregDThomas** [comment]
In the meantime, I've overcome this issue with the following bit of Gradle;

```
    // Until https://github.com/joelittlejohn/jsonschema2pojo/issues/1461 is fixed, correct the package for the "@Generated" annotation
    tasks.named('generateJsonSchema2Pojo') {
        finalizedBy('makeJakartaAnnotationsCompatible')
    }

    tasks.register('makeJakartaAnnotationsCompatible') {
        doLast {
            ant.replaceregexp(match:'javax.annotation.processing.Generated', replace:'jakarta.annotation.Generated', flags:'g', byline:true) {
                fileset(dir: 'build/generated-sources/js2p', includes: '**/*.java')
            }
        }
    }
```

**Comment by darkmastermindz** [comment]
@joelittlejohn after reading #1586 as well, I'm thinking a PR for an `useJakartaAnnotation` to just cover this scenario would be a cleaner solution - thoughts on this proposal?

**Comment by magicDGS** [comment]
Is someboy taking over this issue? It would be great to have a way to fully use jakarta instead of require javax just for the generated annotation. Thanks in advance!

**Comment by sagarshah1983** [comment]
We are also getting blocked on doing java 17+ upgrade on this as we use this plugin heavily and plugin does not have a support to use jakarta annotations yet.
Is there any workaround at minimal?

Appreciate your help and support!

**Comment by darkmastermindz** [comment]
> We are also getting blocked on doing java 17+ upgrade on this as we use this plugin heavily and plugin does not have a support to use jakarta annotations yet. Is there any workaround at minimal?
> 
> Appreciate your help and support!

@sagarshah1983 see [this](https://github.com/joelittlejohn/jsonschema2pojo/issues/1461#issuecomment-1931610804) previous reply from @GregDThomas for the current working workaround.

**Comment by sagarshah1983** [comment]
Thank you for suggesting. My understanding was that it works with gradle
only. Can this be handled with maven as well?
Appreciate your help with this.

**Comment by speedythesnail** [comment]
> Thank you for suggesting. My understanding was that it works with gradle
> only. Can this be handled with maven as well?
> Appreciate your help with this.
> […](#)

CoPilot helped generate this which does something equivelent, update the regex as appropriate, it worked for me and my issue:
```
      <!-- Maven Antrun Plugin for replacing javax.annotation with jakarta.annotation -->
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-antrun-plugin</artifactId>
        <version>1.8</version>
        <executions>
          <execution>
            <id>replace-annotations</id>
            <phase>generate-sources</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <target>
                <replaceregexp match="javax.annotation.processing.Generated"
                               replace="jakarta.annotation.Generated"
                               flags="g">
                  <fileset dir="${project.build.directory}/generated-sources/jsonschema2pojo" includes="**/*.java"/>
                </replaceregexp>
                <replaceregexp match="javax.annotation.Nullable"
                               replace="jakarta.annotation.Nullable"
                               flags="g">
                  <fileset dir="${project.build.directory}/generated-sources/jsonschema2pojo" includes="**/*.java"/>
                </replaceregexp>
                <replaceregexp match="javax.annotation.Nonnull"
                               replace="jakarta.annotation.Nonnull"
                               flags="g">
                  <fileset dir="${project.build.directory}/generated-sources/jsonschema2pojo" includes="**/*.java"/>
                </replaceregexp>
              </target>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

**Comment by sagarshah1983** [comment]
Appreciate this workaround. Thanks for sharing this.

**Comment by ctrimble** [comment]
I have added this as something to take care of for 1.3.0.  There are several of these features nested in the `ObjectRule` 's logic, making them hard to alter when you just need to alter a single feature of the project.

I will work up a PR that moves this to its own `Rule` in the `RuleFactory`.  That would have given a more direct path to fixing this with the `customRuleFactory` option and could be merged in without needing to introduce a new flag.

In a separate PR, we should provide some way to specify the exact `Generated` annotation to include.  I see a few ways to do this:

1. Add `jakarta.annotaiton.Generated` as the first option in the existing search.  This would then trigger the inclusion based on plugin dependencies included by the user.
2. Add a `generatedAnnotation` property that just includes either `auto` or the fully qualified name of the annotation.
3. Add a `useJakartaEe` property that sets everything the right way for Jakarta EE target environments.
